src/main/java/bgu/spl/mics/Future.java

```diff
@@ -43,19 +49,20 @@ public T get() {
      * @pre (isDone () == false)
      * @post (isDone () == true && get() == result)
      */
-    public void resolve(T result) {
-        //TODO: implement this.
+    public synchronized void resolve(T result) {
```

Comment:
why do you need to synchronize this method?

Reply:
Because we want to be able to use wait and notify

Reply:
I think you can use notify without synchronized

Reply:
There are multiple reasons, some are explained [here](https://stackoverflow.com/questions/2779484/why-must-wait-always-be-in-synchronized-block)

Reply:
Yes but they are talking about wait, not notify..

Reply:
explained [here](https://stackoverflow.com/questions/8817266/why-should-the-notify-method-be-inside-a-synchronized-block)